<!DOCTYPE html><html>

<head>
<meta charset="utf-8">
<title>commands_snp</title>
<style type="text/css">
body {
  font-family: Helvetica, arial, sans-serif;
  font-size: 14px;
  line-height: 1.6;
  padding-top: 10px;
  padding-bottom: 10px;
  background-color: white;
  padding: 30px; }

body > *:first-child {
  margin-top: 0 !important; }
body > *:last-child {
  margin-bottom: 0 !important; }

a {
  color: #4183C4; }
a.absent {
  color: #cc0000; }
a.anchor {
  display: block;
  padding-left: 30px;
  margin-left: -30px;
  cursor: pointer;
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0; }

h1, h2, h3, h4, h5, h6 {
  margin: 20px 0 10px;
  padding: 0;
  font-weight: bold;
  -webkit-font-smoothing: antialiased;
  cursor: text;
  position: relative; }

h1:hover a.anchor, h2:hover a.anchor, h3:hover a.anchor, h4:hover a.anchor, h5:hover a.anchor, h6:hover a.anchor {
  background: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA09pVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMy1jMDExIDY2LjE0NTY2MSwgMjAxMi8wMi8wNi0xNDo1NjoyNyAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvIiB4bWxuczp4bXBNTT0iaHR0cDovL25zLmFkb2JlLmNvbS94YXAvMS4wL21tLyIgeG1sbnM6c3RSZWY9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9zVHlwZS9SZXNvdXJjZVJlZiMiIHhtcDpDcmVhdG9yVG9vbD0iQWRvYmUgUGhvdG9zaG9wIENTNiAoMTMuMCAyMDEyMDMwNS5tLjQxNSAyMDEyLzAzLzA1OjIxOjAwOjAwKSAgKE1hY2ludG9zaCkiIHhtcE1NOkluc3RhbmNlSUQ9InhtcC5paWQ6OUM2NjlDQjI4ODBGMTFFMTg1ODlEODNERDJBRjUwQTQiIHhtcE1NOkRvY3VtZW50SUQ9InhtcC5kaWQ6OUM2NjlDQjM4ODBGMTFFMTg1ODlEODNERDJBRjUwQTQiPiA8eG1wTU06RGVyaXZlZEZyb20gc3RSZWY6aW5zdGFuY2VJRD0ieG1wLmlpZDo5QzY2OUNCMDg4MEYxMUUxODU4OUQ4M0REMkFGNTBBNCIgc3RSZWY6ZG9jdW1lbnRJRD0ieG1wLmRpZDo5QzY2OUNCMTg4MEYxMUUxODU4OUQ4M0REMkFGNTBBNCIvPiA8L3JkZjpEZXNjcmlwdGlvbj4gPC9yZGY6UkRGPiA8L3g6eG1wbWV0YT4gPD94cGFja2V0IGVuZD0iciI/PsQhXeAAAABfSURBVHjaYvz//z8DJYCRUgMYQAbAMBQIAvEqkBQWXI6sHqwHiwG70TTBxGaiWwjCTGgOUgJiF1J8wMRAIUA34B4Q76HUBelAfJYSA0CuMIEaRP8wGIkGMA54bgQIMACAmkXJi0hKJQAAAABJRU5ErkJggg==) no-repeat 10px center;
  text-decoration: none; }

h1 tt, h1 code {
  font-size: inherit; }

h2 tt, h2 code {
  font-size: inherit; }

h3 tt, h3 code {
  font-size: inherit; }

h4 tt, h4 code {
  font-size: inherit; }

h5 tt, h5 code {
  font-size: inherit; }

h6 tt, h6 code {
  font-size: inherit; }

h1 {
  font-size: 28px;
  color: black; }

h2 {
  font-size: 24px;
  border-bottom: 1px solid #cccccc;
  color: black; }

h3 {
  font-size: 18px; }

h4 {
  font-size: 16px; }

h5 {
  font-size: 14px; }

h6 {
  color: #777777;
  font-size: 14px; }

p, blockquote, ul, ol, dl, li, table, pre {
  margin: 15px 0; }

hr {
  background: transparent url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAYAAAAECAYAAACtBE5DAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAAyJpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvIiB4bWxuczp4bXBNTT0iaHR0cDovL25zLmFkb2JlLmNvbS94YXAvMS4wL21tLyIgeG1sbnM6c3RSZWY9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9zVHlwZS9SZXNvdXJjZVJlZiMiIHhtcDpDcmVhdG9yVG9vbD0iQWRvYmUgUGhvdG9zaG9wIENTNSBNYWNpbnRvc2giIHhtcE1NOkluc3RhbmNlSUQ9InhtcC5paWQ6OENDRjNBN0E2NTZBMTFFMEI3QjRBODM4NzJDMjlGNDgiIHhtcE1NOkRvY3VtZW50SUQ9InhtcC5kaWQ6OENDRjNBN0I2NTZBMTFFMEI3QjRBODM4NzJDMjlGNDgiPiA8eG1wTU06RGVyaXZlZEZyb20gc3RSZWY6aW5zdGFuY2VJRD0ieG1wLmlpZDo4Q0NGM0E3ODY1NkExMUUwQjdCNEE4Mzg3MkMyOUY0OCIgc3RSZWY6ZG9jdW1lbnRJRD0ieG1wLmRpZDo4Q0NGM0E3OTY1NkExMUUwQjdCNEE4Mzg3MkMyOUY0OCIvPiA8L3JkZjpEZXNjcmlwdGlvbj4gPC9yZGY6UkRGPiA8L3g6eG1wbWV0YT4gPD94cGFja2V0IGVuZD0iciI/PqqezsUAAAAfSURBVHjaYmRABcYwBiM2QSA4y4hNEKYDQxAEAAIMAHNGAzhkPOlYAAAAAElFTkSuQmCC) repeat-x 0 0;
  border: 0 none;
  color: #cccccc;
  height: 4px;
  padding: 0;
}

body > h2:first-child {
  margin-top: 0;
  padding-top: 0; }
body > h1:first-child {
  margin-top: 0;
  padding-top: 0; }
  body > h1:first-child + h2 {
    margin-top: 0;
    padding-top: 0; }
body > h3:first-child, body > h4:first-child, body > h5:first-child, body > h6:first-child {
  margin-top: 0;
  padding-top: 0; }

a:first-child h1, a:first-child h2, a:first-child h3, a:first-child h4, a:first-child h5, a:first-child h6 {
  margin-top: 0;
  padding-top: 0; }

h1 p, h2 p, h3 p, h4 p, h5 p, h6 p {
  margin-top: 0; }

li p.first {
  display: inline-block; }
li {
  margin: 0; }
ul, ol {
  padding-left: 30px; }

ul :first-child, ol :first-child {
  margin-top: 0; }

dl {
  padding: 0; }
  dl dt {
    font-size: 14px;
    font-weight: bold;
    font-style: italic;
    padding: 0;
    margin: 15px 0 5px; }
    dl dt:first-child {
      padding: 0; }
    dl dt > :first-child {
      margin-top: 0; }
    dl dt > :last-child {
      margin-bottom: 0; }
  dl dd {
    margin: 0 0 15px;
    padding: 0 15px; }
    dl dd > :first-child {
      margin-top: 0; }
    dl dd > :last-child {
      margin-bottom: 0; }

blockquote {
  border-left: 4px solid #dddddd;
  padding: 0 15px;
  color: #777777; }
  blockquote > :first-child {
    margin-top: 0; }
  blockquote > :last-child {
    margin-bottom: 0; }

table {
  padding: 0;border-collapse: collapse; }
  table tr {
    border-top: 1px solid #cccccc;
    background-color: white;
    margin: 0;
    padding: 0; }
    table tr:nth-child(2n) {
      background-color: #f8f8f8; }
    table tr th {
      font-weight: bold;
      border: 1px solid #cccccc;
      margin: 0;
      padding: 6px 13px; }
    table tr td {
      border: 1px solid #cccccc;
      margin: 0;
      padding: 6px 13px; }
    table tr th :first-child, table tr td :first-child {
      margin-top: 0; }
    table tr th :last-child, table tr td :last-child {
      margin-bottom: 0; }

img {
  max-width: 100%; }

span.frame {
  display: block;
  overflow: hidden; }
  span.frame > span {
    border: 1px solid #dddddd;
    display: block;
    float: left;
    overflow: hidden;
    margin: 13px 0 0;
    padding: 7px;
    width: auto; }
  span.frame span img {
    display: block;
    float: left; }
  span.frame span span {
    clear: both;
    color: #333333;
    display: block;
    padding: 5px 0 0; }
span.align-center {
  display: block;
  overflow: hidden;
  clear: both; }
  span.align-center > span {
    display: block;
    overflow: hidden;
    margin: 13px auto 0;
    text-align: center; }
  span.align-center span img {
    margin: 0 auto;
    text-align: center; }
span.align-right {
  display: block;
  overflow: hidden;
  clear: both; }
  span.align-right > span {
    display: block;
    overflow: hidden;
    margin: 13px 0 0;
    text-align: right; }
  span.align-right span img {
    margin: 0;
    text-align: right; }
span.float-left {
  display: block;
  margin-right: 13px;
  overflow: hidden;
  float: left; }
  span.float-left span {
    margin: 13px 0 0; }
span.float-right {
  display: block;
  margin-left: 13px;
  overflow: hidden;
  float: right; }
  span.float-right > span {
    display: block;
    overflow: hidden;
    margin: 13px auto 0;
    text-align: right; }

code, tt {
  margin: 0 2px;
  padding: 0 5px;
  white-space: nowrap;
  border: 1px solid #eaeaea;
  background-color: #f8f8f8;
  border-radius: 3px; }

pre code {
  margin: 0;
  padding: 0;
  white-space: pre;
  border: none;
  background: transparent; }

.highlight pre {
  background-color: #f8f8f8;
  border: 1px solid #cccccc;
  font-size: 13px;
  line-height: 19px;
  overflow: auto;
  padding: 6px 10px;
  border-radius: 3px; }

pre {
  background-color: #f8f8f8;
  border: 1px solid #cccccc;
  font-size: 13px;
  line-height: 19px;
  overflow: auto;
  padding: 6px 10px;
  border-radius: 3px; }
  pre code, pre tt {
    background-color: transparent;
    border: none; }

sup {
    font-size: 0.83em;
    vertical-align: super;
    line-height: 0;
}
* {
	-webkit-print-color-adjust: exact;
}
@media screen and (min-width: 914px) {
    body {
        width: 854px;
        margin:0 auto;
    }
}
@media print {
	table, pre {
		page-break-inside: avoid;
	}
	pre {
		word-wrap: break-word;
	}
}
</style>
<style type="text/css">
/**
 * prism.js default theme for JavaScript, CSS and HTML
 * Based on dabblet (http://dabblet.com)
 * @author Lea Verou
 */

code[class*="language-"],
pre[class*="language-"] {
	color: black;
	text-shadow: 0 1px white;
	font-family: Consolas, Monaco, 'Andale Mono', monospace;
	direction: ltr;
	text-align: left;
	white-space: pre;
	word-spacing: normal;
	word-break: normal;
	line-height: 1.5;

	-moz-tab-size: 4;
	-o-tab-size: 4;
	tab-size: 4;

	-webkit-hyphens: none;
	-moz-hyphens: none;
	-ms-hyphens: none;
	hyphens: none;
}

pre[class*="language-"]::-moz-selection, pre[class*="language-"] ::-moz-selection,
code[class*="language-"]::-moz-selection, code[class*="language-"] ::-moz-selection {
	text-shadow: none;
	background: #b3d4fc;
}

pre[class*="language-"]::selection, pre[class*="language-"] ::selection,
code[class*="language-"]::selection, code[class*="language-"] ::selection {
	text-shadow: none;
	background: #b3d4fc;
}

@media print {
	code[class*="language-"],
	pre[class*="language-"] {
		text-shadow: none;
	}
}

/* Code blocks */
pre[class*="language-"] {
	padding: 1em;
	margin: .5em 0;
	overflow: auto;
}

:not(pre) > code[class*="language-"],
pre[class*="language-"] {
	background: #f5f2f0;
}

/* Inline code */
:not(pre) > code[class*="language-"] {
	padding: .1em;
	border-radius: .3em;
}

.token.comment,
.token.prolog,
.token.doctype,
.token.cdata {
	color: slategray;
}

.token.punctuation {
	color: #999;
}

.namespace {
	opacity: .7;
}

.token.property,
.token.tag,
.token.boolean,
.token.number,
.token.constant,
.token.symbol,
.token.deleted {
	color: #905;
}

.token.selector,
.token.attr-name,
.token.string,
.token.char,
.token.builtin,
.token.inserted {
	color: #690;
}

.token.operator,
.token.entity,
.token.url,
.language-css .token.string,
.style .token.string {
	color: #a67f59;
	background: hsla(0, 0%, 100%, .5);
}

.token.atrule,
.token.attr-value,
.token.keyword {
	color: #07a;
}

.token.function {
	color: #DD4A68;
}

.token.regex,
.token.important,
.token.variable {
	color: #e90;
}

.token.important,
.token.bold {
	font-weight: bold;
}
.token.italic {
	font-style: italic;
}

.token.entity {
	cursor: help;
}
</style>
</head>
<body>
<h1 id="toc_0">SNP detection Tutorials - 2015 - v3.0.0</h1>

<p>Next Gen Sequencing Training Materials for the Bioinformatics Resource Center
<br>Biotechnology Center, 425 Henry Mall, Univeristy of Wisconsin-Madison, USA.</p>

<blockquote>
<p>Jean-Yves Sgro,
Jeremy Glasner, &amp; Xiao-yu Liu</p>
</blockquote>

<h1 id="toc_1">WORKSHOP COMMANDS</h1>

<p>All (or most) of the commands used in the workbook are presented again here as a summary. Explanations should be sought within the workbook itself.</p>

<p>The HTML or PDF version(s) of this document, when available in class, allows you to copy/paste all the commands into your Terminal session. However, please heed the following recommendation:</p>

<p>COMMANDS ARE PRESENTED WITHOUT THE <code>$</code> PROMPT FOR EASY COPY/PASTE OF MULTIPLE LINES. HOWEVER, IT IS RECOMMENDED TO COPY/PASTE ONLY ONE LINE AT A TIME FOR COMPLEX CASES TO SEE AND UNDERSTAND THE PURPOSE OF THE COMMAND(S).</p>

<h2 id="toc_2">Connect to BRC from Mac or Windows</h2>

<ul>
<li>On Mac use Terminal</li>
<li>On Windows use MobaXterm</li>
</ul>

<h3 id="toc_3">Remote connection with ssh</h3>

<blockquote>
<p>REPLACE <em>brcuser</em> and <em>remote-machine</em> with your credentials</p>
</blockquote>

<pre><code class="language-bash">ssh -Y brcuser@remote-machine.ad.biotech.wisc.edu</code></pre>

<h2 id="toc_4">NEXT GEN WORKSHOP COMMANDS</h2>

<h3 id="toc_5">BRC TERMINAL / BRC SET-UP</h3>

<pre><code class="language-bash"> alias ls
 alias ll
 ls -ld /mnt/software/bow* | cut -c 20-120</code></pre>

<pre><code class="language-bash"> which bowtie
 bowtie # THEN PRESS [TAB][TAB]</code></pre>

<h3 id="toc_6">SET-UP</h3>

<pre><code class="language-bash"> pwd
 ls -l
 mkdir SNP1
 ls -lF
 cd SNP1</code></pre>

<h3 id="toc_7">INDEXED GENOMES</h3>

<pre><code class="language-bash"> ls -l /mnt/software/bowtie-indexes/hg_19* | cut -c 29-100</code></pre>

<pre><code class="language-bash"> printenv BOWTIE_INDEXES
 export BOWTIE_INDEXES=/mnt/software/bowtie-indexes
 printenv BOWTIE_INDEXES</code></pre>

<pre><code class="language-bash"> ls /mnt/software/workshop/genomes/hg19/genome.fa*
 pwd
 ln -s /mnt/software/workshop/genomes/hg19/genome.fa  genome.fa
 ls  -l
 cp /mnt/software/workshop/genomes/hg19/genome.fa.fai genome.fa.fai
 ls </code></pre>

<h3 id="toc_8">COPY FQ READ FILE (FROM WITHIN YOUR SNP1 DIRECTORY)</h3>

<pre><code class="language-bash"> pwd
 cp /mnt/software/workshop/data/chr20_5.fq  .
 ls </code></pre>

<h3 id="toc_9">MAP WITH BOWTIE</h3>

<pre><code class="language-bash"> which bowtie</code></pre>

<pre><code class="language-bash"># bowtie -a -m 3 -S hg_19 chr20_5.fq chr20_5_aln.sam</code></pre>

<pre><code class="language-bash"># Use the multiprocessor with 3 processors if possible:
 bowtie -a -m 3 -p 2 -S hg_19 chr20_5.fq chr20_5_aln.sam</code></pre>

<pre><code class="language-bash"> head -3 chr20_5_aln.sam</code></pre>

<h3 id="toc_10">CONTINUE MAPPING WITH SAMTOOLS</h3>

<pre><code class="language-bash"> which samtools
 samtools
 samtools view
 ls -l genome.fa*</code></pre>

<pre><code class="language-bash"> samtools import genome.fa.fai chr20_5_aln.sam chr20_5_aln.bam
 samtools sort chr20_5_aln.bam chr20_5_aln.sorted
 ls -t chr20*
 samtools index chr20_5_aln.sorted.bam
 ls -1t chr20*</code></pre>

<blockquote>
<p>BOX INFO</p>

<pre><code class="language-bash">ln -s ../Homo_sapiens/UCSC/hg19/Sequence/Chromosomes/chr20.fa chr20.fa</code></pre>
</blockquote>

<h3 id="toc_11">INTERACTIVE TXT VIEWER:</h3>

<pre><code class="language-bash"> samtools tview chr20_5_aln.sorted.bam genome.fa</code></pre>

<h3 id="toc_12">CALL SNPs</h3>

<pre><code class="language-bash"> which bcftools
 bcftools --version
 samtools mpileup</code></pre>

<p>The following command works with samtools 0.1.19, for newer version e.g. 1.1.0  see box below.</p>

<pre><code class="language-bash"> samtools mpileup -uD -f genome.fa chr20_5_aln.sorted.bam | bcftools view -bvcg -  &gt; chr20_5_samtools.raw.bcf</code></pre>

<blockquote>
<p>Note:
NEW COMMAND: create VCF directly (VCF4.2 format)</p>

<p>ALSO in <code>samtools</code>: <code>-D</code> is deprecated =&gt; use <code>-t DP</code></p>

<p>FULL PATH MAY BE REQUIRED DEPENDING ON --version</p>
</blockquote>

<p><code>/mnt/software/samtools-dir/samtools-1.1/bin/samtools mpileup -u -t DP -f genome.fa chr20_5_aln.sorted.bam | bcftools view -v snps,indels,mnps &gt; chr20_5_samtools.raw.vcf</code></p>

<blockquote>
<p>The following command may also work:</p>
</blockquote>

<p><code>/mnt/software/samtools-dir/samtools-1.1/bin/samtools mpileup -u -t DP -f genome.fa chr20_5_aln.sorted.bam | bcftools call -cv - &gt; out-put_2.vcf</code></p>

<blockquote>
</blockquote>

<h3 id="toc_13">CONVERT BCF TO VCF</h3>

<pre><code class="language-bash"> bcftools view chr20_5_samtools.raw.bcf | /mnt/software/samtools/bcftools/vcfutils.pl varFilter -D100 &gt; chr20_5_samtools.vcf</code></pre>

<pre><code class="language-bash"> ls -lt *vcf
 head -3 chr20_5_samtools.vcf
 less chr20_5_samtools.vcf
 tail -3 chr20_5_samtools.vcf
 grep -v chr20 chr20_5_samtools.vcf | wc</code></pre>

<blockquote>
<h3 id="toc_14">Quick Editing of VCF file with &quot;vi&quot; editor:</h3>

<p>This is no longer required with the newest samtools/bcftools versions creating VCF 4.2 or greater file versions. However, this is a very fancy &quot;in-line&quot; use of the <code>vi</code> editor worth knowing....
<code>sh
vi chr20_5_samtools.vcf &quot;+:%s/Number=-1/Number=1/g&quot; &quot;+:wq&quot;
</code></p>
</blockquote>

<h3 id="toc_15">TRANSFER FILES TO YOUR COMPUTER</h3>

<p>Make local directory called BAMS</p>

<pre><code class="language-bash">cd $HOME/Desktop/BAMS
pwd</code></pre>

<p>Then connect to remote computer:</p>

<ul>
<li>replace <code>brcuser</code> with your username</li>
<li>replace <code>remote-machine</code> with the actual name, <em>e.g.</em> <code>sumo</code> </li>
<li>for example:
<code>sftp brcuser@remote-machine.ad.biotech.wisc.edu</code></li>
</ul>

<h3 id="toc_16">WITHIN SFTP, commands:</h3>

<pre><code class="language-bash">cd SNP1
ls *sorted*
mget *sorted*bam*
exit</code></pre>

<h3 id="toc_17">IGV GENOME VIEWER</h3>

<p>RUN ON LOCAL COMPUTER</p>

<h3 id="toc_18">SNPEFF COMMAND TYPE IN ONE LINES</h3>

<pre><code class="language-bash">java -Xmx2g -jar /mnt/software/workshop/software/snpEff/snpEff.jar -c /mnt/software/workshop/software/snpEff/snpEff.config -snp -v hg19 chr20_5_samtools.vcf  &gt;  chr20_5_samtools_snp.vcf</code></pre>

<h3 id="toc_19">COUNTING EFFECTS WITHIN ANNOTATED VCF FILE:</h3>

<pre><code class="language-bash"> fgrep ID=EFF chr20_5_samtools_snp.vcf 
 fgrep START_LOST chr20_5_samtools_snp.vcf
 fgrep START_LOST chr20_5_samtools_snp.vcf |wc
 fgrep START_GAIN chr20_5_samtools_snp.vcf |wc
 fgrep STOP_GAIN chr20_5_samtools_snp.vcf |wc
 fgrep STOP_LOST chr20_5_samtools_snp.vcf |wc
 fgrep SPLICE_SITE_ACCEPTOR chr20_5_samtools_snp.vcf |wc
 fgrep HIGH chr20_5_samtools_snp.vcf |wc
 fgrep MODERATE chr20_5_samtools_snp.vcf |wc
 fgrep LOW chr20_5_samtools_snp.vcf |wc</code></pre>

<pre><code class="language-bash"> firefox snpEff_summary.html &amp;</code></pre>

<h3 id="toc_20">SFTP FILES TO LOCAL COMPUTER</h3>

<pre><code class="language-bash"> cd $HOME/Desktop</code></pre>

<p>Example:
<code>sftp brcuser@remote-machine.ad.biotech.wisc.edu</code></p>

<h3 id="toc_21">WITHIN SFTP SESSION:</h3>

<pre><code class="language-bash"> cd SNP1
 ls –t
 get snpEff_summary.html
 get snpEff_genes.txt
 get chr20_5_samtools_snp.vcf
exit</code></pre>

<h2 id="toc_22">PART 3 CASE STUDY COMMANDS</h2>

<blockquote>
<p>NOTE: COPY (<code>cp</code>) commands below for “pre-run data” are 
commented (<code>#</code>) by default. 
If you want to actually copy the file(s) do NOT
type the <code>#</code> character itself.</p>
</blockquote>

<p>The summary web page is at: 
<a href="http://www.ncbi.nlm.nih.gov/sra/?term=SRR957857">http://www.ncbi.nlm.nih.gov/sra/?term=SRR957857</a></p>

<h3 id="toc_23">Create new directory for this project</h3>

<pre><code class="language-bash"> cd $HOME
 mkdir SNP-sup
 cd SNP-sup</code></pre>

<pre><code class="language-bash"> mkdir genome
 mkdir fastq
 mkdir prerun</code></pre>

<blockquote>
<p>example command
 <code>#  cp file_XYZ_123 ./prerun</code></p>
</blockquote>

<h3 id="toc_24">Download the Data</h3>

<pre><code class="language-bash"> wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR957/SRR957857/SRR957857_1.fastq.gz</code></pre>

<pre><code class="language-bash"> wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR957/SRR957857/SRR957857_2.fastq.gz</code></pre>

<blockquote>
<p>OR: Copy pre-run:
<code># cp /mnt/software/workshop/data/SNP-sup/SRR957857_*.gz  fastq</code></p>
</blockquote>

<pre><code class="language-bash"> zcat SRR957857_1.fastq.gz | head -4</code></pre>

<pre><code class="language-bash"> zcat SRR957857_2.fastq.gz | head -2 |tail -1 | wc</code></pre>

<pre><code class="language-bash">wget -O CP008957.fasta &quot;http://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=nucleotide&amp;id=CP008957.1&amp;rettype=fasta&amp;retmode=txt&quot;</code></pre>

<pre><code class="language-bash">wget -O CP008958.fasta &quot;http://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=nucleotide&amp;id=CP008958.1&amp;rettype=fasta&amp;retmode=txt&quot;</code></pre>

<h3 id="toc_25">BOX</h3>

<p><code># curl -s &quot;http://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=nucleotide&amp;id=CP008957.1&amp;rettype=fasta&amp;retmode=txt&quot; &gt; CP008957.fasta</code></p>

<p><code># curl -s &quot;http://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=nucleotide&amp;id=CP008958.1&amp;rettype=fasta&amp;retmode=txt&quot; &gt; CP008958.fasta</code></p>

<p><code># cp /mnt/software/workshop/data/SNP-sup/CP00895*.fasta  genome</code></p>

<h3 id="toc_26">TASK:</h3>

<pre><code class="language-bash">cat genome/CP008958.fasta genome/CP008957.fasta &gt; genome/EDL933.fasta</code></pre>

<pre><code class="language-bash">grep &quot;&gt;&quot; genome/EDL933.fasta</code></pre>

<h3 id="toc_27">Map reads to reference genome with bwa</h3>

<pre><code class="language-bash">bwa index EDL933.fasta</code></pre>

<pre><code class="language-bash">bwa mem -t 4 genome/EDL933.fasta fastq/SRR957857_1.fastq.gz fastq/SRR957857_2.fastq.gz &gt; SRR957857.sam</code></pre>

<pre><code class="language-bash">head -5 SRR957857.sam
```sh</code></pre>

<p>head -4 SRR957857.sam | cat -b
<code>
</code>sh
ls -lh *sam
```</p>

<h3 id="toc_28">Reorganize SAM into sorted, indexed BAM</h3>

<pre><code class="language-bash">samtools view -b -S SRR957857.sam &gt; SRR957857.bam</code></pre>

<pre><code class="language-bash">ls –lh *.bam</code></pre>

<pre><code class="language-bash">samtools sort SRR957857.bam SRR957857_sort</code></pre>

<pre><code class="language-bash">samtools index SRR957857_sort.bam</code></pre>

<h3 id="toc_29">Identify SNPs</h3>

<pre><code class="language-bash">/mnt/software/freebayes/freebayes</code></pre>

<pre><code class="language-bash">/mnt/software/freebayes/freebayes -f genome/EDL933.fasta SRR957857_sort.bam &gt; SRR957857.vcf</code></pre>

<blockquote>
<p>Or Copy prerun:
<code># cp /mnt/software/workshop/data/SNP-sup/prerun/SRR957857.vcf  .</code></p>
</blockquote>

<pre><code class="language-bash">cat &gt; pindel_config.txt
SRR957857_sort.bam      338     SRR957857_sample</code></pre>

<blockquote>
<p>Dont&#39; forget: CTRL-D TO EXIT cat</p>
</blockquote>

<pre><code class="language-bash">pindel</code></pre>

<pre><code class="language-bash">pindel -f /genome/EDL933.fasta -i pindel_config.txt -o SRR957857</code></pre>

<blockquote>
<p>Or copy prerun:
<code># cp /mnt/software/workshop/data/SNP-sup/prerun/prerun_pindel_outputs/* ./prerun</code></p>
</blockquote>

<pre><code class="language-bash">cat SRR957857_D | grep &quot;&lt;&quot;</code></pre>

<pre><code class="language-bash">cat SRR957857_D | fgrep &quot;&lt;&quot; | sed -e &#39;s/&lt;/ /g&#39; -e &#39;s/&gt;/ /g&#39;| cut -f 2 -d &quot; &quot; | sort -n</code></pre>

<pre><code class="language-bash">pindel2vcf -p SRR957857_D -r genome/EDL933.fasta  -R EDL933v3 -d 20140711</code></pre>

<h3 id="toc_30">BOX</h3>

<blockquote>
<pre><code class="language-bash"># for f in *.vcf; do pindel2vcf -p $f -r genome/EDL933.fasta  -R EDL933v3 -d 20140711; done</code></pre>
</blockquote>

<h3 id="toc_31">Annotate Variants</h3>

<h4 id="toc_32">Build a snpEff database</h4>

<pre><code class="language-bash"> mkdir snpEff
 cd snpEff
 mkdir data
 cd data
 mkdir EDL933
 cd EDL933</code></pre>

<blockquote>
<p>Or copy prerun:</p>

<pre><code class="language-bash"># cp /mnt/software/workshop/data/SNP-sup/prerun/data/EDL933/genes.gff .</code></pre>
</blockquote>

<pre><code class="language-bash">cd ../..</code></pre>

<p>(If in doubt verify that you are back within <code>~/SNP-sup/snpEff</code> directory with <code>pwd</code>.)</p>

<blockquote>
<p>Or copy prerun:</p>

<pre><code class="language-ssh"> # cp /mnt/software/workshop/data/SNP-sup/prerun/snpEff.config .</code></pre>

<pre><code class="language-bash"> # cp /mnt/software/workshop/data/SNP-sup/prerun/snpEff.jar . </code></pre>
</blockquote>

<pre><code class="language-bash">java -jar snpEff.jar build -gff3 -v EDL933</code></pre>

<h4 id="toc_33">Annotate Variants with snpEff</h4>

<pre><code class="language-bash">java -jar ./snpEff/snpEff.jar -no-upstream -no-downstream  -no-intergenic EDL933 SRR957857.vcf &gt; SRR957857_snpEff_noupdowninter.vcf</code></pre>

<pre><code class="language-bash">grep frameshift SRR957857_snpEff_noupdowninter.vcf | more</code></pre>

<pre><code class="language-bash">java -jar ./snpEff/snpEff.jar -no-upstream -no-downstream -no-intergenic EDL933 SRR957857_D.vcf &gt; SRR957857_D_snpEff_noupdowninter.vcf</code></pre>

<h3 id="toc_34">Visualize Results</h3>

<p>Transfer files and use local IGV.</p>

<h3 id="toc_35">Compare Multiple Genomes</h3>

<pre><code class="language-bash"> cd $HOME/SNP-sup</code></pre>

<pre><code class="language-bash"> mkdir alignments
 cd alignments</code></pre>

<pre><code class="language-bash">unzip /mnt/software/workshop/data/SNP-sup/O157_alignment.zip</code></pre>

<pre><code class="language-bash">which mauve</code></pre>

<pre><code class="language-bash">mauve 6_o157_align_ro &amp;</code></pre>

<h1 id="toc_36">APPENDIX COMMANDS</h1>

<h2 id="toc_37">SIMULATED DATA:</h2>

<pre><code class="language-bash"> seqret -sbegin 499950 -send 3651230
 wc chr20_5.fa 
 gzip chr20_5.fa 
 wgsim -N 200000 -r 0.01 -1 100 -S11 -d0 -e0 
 head chr20_5.fq</code></pre>

<h2 id="toc_38">GENOMES</h2>

<p>Uncomment to use the next lines!</p>

<pre><code class="language-bash"># wget ftp://igenome:G3nom3s4u@ussd-ftp.illumina.com/Homo_sapiens/Ensembl/GRCh37/Homo_sapiens_Ensembl_GRCh37.tar.gz</code></pre>

<pre><code class="language-bash"># wget ftp://igenome:G3nom3s4u@ussd-ftp.illumina.com/Homo_sapiens/UCSC/hg19/Homo_sapiens_UCSC_hg19.tar.gz</code></pre>

<pre><code class="language-bash"># wget ftp://igenome:G3nom3s4u@ussd-ftp.illumina.com/Homo_sapiens/NCBI/build37.2/Homo_sapiens_NCBI_build37.2.tar.gz</code></pre>

<h1 id="toc_39">END COMMANDS</h1>

<script type="text/javascript">
self="undefined"!=typeof window?window:"undefined"!=typeof WorkerGlobalScope&&self instanceof WorkerGlobalScope?self:{};var Prism=function(){var e=/\blang(?:uage)?-(?!\*)(\w+)\b/i,t=self.Prism={util:{encode:function(e){return e instanceof n?new n(e.type,t.util.encode(e.content),e.alias):"Array"===t.util.type(e)?e.map(t.util.encode):e.replace(/&/g,"&amp;").replace(/</g,"&lt;").replace(/\u00a0/g," ")},type:function(e){return Object.prototype.toString.call(e).match(/\[object (\w+)\]/)[1]},clone:function(e){var n=t.util.type(e);switch(n){case"Object":var a={};for(var r in e)e.hasOwnProperty(r)&&(a[r]=t.util.clone(e[r]));return a;case"Array":return e.map(function(e){return t.util.clone(e)})}return e}},languages:{extend:function(e,n){var a=t.util.clone(t.languages[e]);for(var r in n)a[r]=n[r];return a},insertBefore:function(e,n,a,r){r=r||t.languages;var i=r[e];if(2==arguments.length){a=arguments[1];for(var l in a)a.hasOwnProperty(l)&&(i[l]=a[l]);return i}var s={};for(var o in i)if(i.hasOwnProperty(o)){if(o==n)for(var l in a)a.hasOwnProperty(l)&&(s[l]=a[l]);s[o]=i[o]}return t.languages.DFS(t.languages,function(t,n){n===r[e]&&t!=e&&(this[t]=s)}),r[e]=s},DFS:function(e,n,a){for(var r in e)e.hasOwnProperty(r)&&(n.call(e,r,e[r],a||r),"Object"===t.util.type(e[r])?t.languages.DFS(e[r],n):"Array"===t.util.type(e[r])&&t.languages.DFS(e[r],n,r))}},highlightAll:function(e,n){for(var a,r=document.querySelectorAll('code[class*="language-"], [class*="language-"] code, code[class*="lang-"], [class*="lang-"] code'),i=0;a=r[i++];)t.highlightElement(a,e===!0,n)},highlightElement:function(a,r,i){for(var l,s,o=a;o&&!e.test(o.className);)o=o.parentNode;if(o&&(l=(o.className.match(e)||[,""])[1],s=t.languages[l]),s){a.className=a.className.replace(e,"").replace(/\s+/g," ")+" language-"+l,o=a.parentNode,/pre/i.test(o.nodeName)&&(o.className=o.className.replace(e,"").replace(/\s+/g," ")+" language-"+l);var u=a.textContent;if(u){u=u.replace(/^(?:\r?\n|\r)/,"");var g={element:a,language:l,grammar:s,code:u};if(t.hooks.run("before-highlight",g),r&&self.Worker){var c=new Worker(t.filename);c.onmessage=function(e){g.highlightedCode=n.stringify(JSON.parse(e.data),l),t.hooks.run("before-insert",g),g.element.innerHTML=g.highlightedCode,i&&i.call(g.element),t.hooks.run("after-highlight",g)},c.postMessage(JSON.stringify({language:g.language,code:g.code}))}else g.highlightedCode=t.highlight(g.code,g.grammar,g.language),t.hooks.run("before-insert",g),g.element.innerHTML=g.highlightedCode,i&&i.call(a),t.hooks.run("after-highlight",g)}}},highlight:function(e,a,r){var i=t.tokenize(e,a);return n.stringify(t.util.encode(i),r)},tokenize:function(e,n){var a=t.Token,r=[e],i=n.rest;if(i){for(var l in i)n[l]=i[l];delete n.rest}e:for(var l in n)if(n.hasOwnProperty(l)&&n[l]){var s=n[l];s="Array"===t.util.type(s)?s:[s];for(var o=0;o<s.length;++o){var u=s[o],g=u.inside,c=!!u.lookbehind,f=0,h=u.alias;u=u.pattern||u;for(var p=0;p<r.length;p++){var d=r[p];if(r.length>e.length)break e;if(!(d instanceof a)){u.lastIndex=0;var m=u.exec(d);if(m){c&&(f=m[1].length);var y=m.index-1+f,m=m[0].slice(f),v=m.length,k=y+v,b=d.slice(0,y+1),w=d.slice(k+1),N=[p,1];b&&N.push(b);var O=new a(l,g?t.tokenize(m,g):m,h);N.push(O),w&&N.push(w),Array.prototype.splice.apply(r,N)}}}}}return r},hooks:{all:{},add:function(e,n){var a=t.hooks.all;a[e]=a[e]||[],a[e].push(n)},run:function(e,n){var a=t.hooks.all[e];if(a&&a.length)for(var r,i=0;r=a[i++];)r(n)}}},n=t.Token=function(e,t,n){this.type=e,this.content=t,this.alias=n};if(n.stringify=function(e,a,r){if("string"==typeof e)return e;if("Array"===t.util.type(e))return e.map(function(t){return n.stringify(t,a,e)}).join("");var i={type:e.type,content:n.stringify(e.content,a,r),tag:"span",classes:["token",e.type],attributes:{},language:a,parent:r};if("comment"==i.type&&(i.attributes.spellcheck="true"),e.alias){var l="Array"===t.util.type(e.alias)?e.alias:[e.alias];Array.prototype.push.apply(i.classes,l)}t.hooks.run("wrap",i);var s="";for(var o in i.attributes)s+=o+'="'+(i.attributes[o]||"")+'"';return"<"+i.tag+' class="'+i.classes.join(" ")+'" '+s+">"+i.content+"</"+i.tag+">"},!self.document)return self.addEventListener?(self.addEventListener("message",function(e){var n=JSON.parse(e.data),a=n.language,r=n.code;self.postMessage(JSON.stringify(t.util.encode(t.tokenize(r,t.languages[a])))),self.close()},!1),self.Prism):self.Prism;var a=document.getElementsByTagName("script");return a=a[a.length-1],a&&(t.filename=a.src,document.addEventListener&&!a.hasAttribute("data-manual")&&document.addEventListener("DOMContentLoaded",t.highlightAll)),self.Prism}();"undefined"!=typeof module&&module.exports&&(module.exports=Prism);
</script>
<script type="text/javascript">
Prism.languages.clike={comment:[{pattern:/(^|[^\\])\/\*[\w\W]*?\*\//,lookbehind:!0},{pattern:/(^|[^\\:])\/\/.+/,lookbehind:!0}],string:/("|')(\\\n|\\?.)*?\1/,"class-name":{pattern:/((?:(?:class|interface|extends|implements|trait|instanceof|new)\s+)|(?:catch\s+\())[a-z0-9_\.\\]+/i,lookbehind:!0,inside:{punctuation:/(\.|\\)/}},keyword:/\b(if|else|while|do|for|return|in|instanceof|function|new|try|throw|catch|finally|null|break|continue)\b/,"boolean":/\b(true|false)\b/,"function":{pattern:/[a-z0-9_]+\(/i,inside:{punctuation:/\(/}},number:/\b-?(0x[\dA-Fa-f]+|\d*\.?\d+([Ee]-?\d+)?)\b/,operator:/[-+]{1,2}|!|<=?|>=?|={1,3}|&{1,2}|\|?\||\?|\*|\/|~|\^|%/,ignore:/&(lt|gt|amp);/i,punctuation:/[{}[\];(),.:]/};
</script>
<script type="text/javascript">
Prism.languages.bash=Prism.languages.extend("clike",{comment:{pattern:/(^|[^"{\\])(#.*?(\r?\n|$))/,lookbehind:!0},string:{pattern:/("|')(\\?[\s\S])*?\1/,inside:{property:/\$([a-zA-Z0-9_#\?\-\*!@]+|\{[^\}]+\})/}},number:{pattern:/([^\w\.])-?(0x[\dA-Fa-f]+|\d*\.?\d+([Ee]-?\d+)?)\b/,lookbehind:!0},"function":/\b(?:alias|apropos|apt-get|aptitude|aspell|awk|basename|bash|bc|bg|builtin|bzip2|cal|cat|cd|cfdisk|chgrp|chmod|chown|chroot|chkconfig|cksum|clear|cmp|comm|command|cp|cron|crontab|csplit|cut|date|dc|dd|ddrescue|declare|df|diff|diff3|dig|dir|dircolors|dirname|dirs|dmesg|du|echo|egrep|eject|enable|env|ethtool|eval|exec|exit|expand|expect|export|expr|fdformat|fdisk|fg|fgrep|file|find|fmt|fold|format|free|fsck|ftp|fuser|gawk|getopts|git|grep|groupadd|groupdel|groupmod|groups|gzip|hash|head|help|hg|history|hostname|htop|iconv|id|ifconfig|ifdown|ifup|import|install|jobs|join|kill|killall|less|link|ln|locate|logname|logout|look|lpc|lpr|lprint|lprintd|lprintq|lprm|ls|lsof|make|man|mkdir|mkfifo|mkisofs|mknod|more|most|mount|mtools|mtr|mv|mmv|nano|netstat|nice|nl|nohup|notify-send|nslookup|open|op|passwd|paste|pathchk|ping|pkill|popd|pr|printcap|printenv|printf|ps|pushd|pv|pwd|quota|quotacheck|quotactl|ram|rar|rcp|read|readarray|readonly|reboot|rename|renice|remsync|rev|rm|rmdir|rsync|screen|scp|sdiff|sed|select|seq|service|sftp|shift|shopt|shutdown|sleep|slocate|sort|source|split|ssh|stat|strace|su|sudo|sum|suspend|sync|tail|tar|tee|test|time|timeout|times|touch|top|traceroute|trap|tr|tsort|tty|type|ulimit|umask|umount|unalias|uname|unexpand|uniq|units|unrar|unshar|until|uptime|useradd|userdel|usermod|users|uuencode|uudecode|v|vdir|vi|vmstat|wait|watch|wc|wget|whereis|which|who|whoami|write|xargs|xdg-open|yes|zip)\b/,keyword:/\b(if|then|else|elif|fi|for|break|continue|while|in|case|function|select|do|done|until|echo|exit|return|set|declare)\b/}),Prism.languages.insertBefore("bash","keyword",{property:/\$([a-zA-Z0-9_#\?\-\*!@]+|\{[^}]+\})/}),Prism.languages.insertBefore("bash","comment",{important:/(^#!\s*\/bin\/bash)|(^#!\s*\/bin\/sh)/});
</script>
</body>

</html>
